packages/apollo-server-core/src/plugin/schemaReporting/index.ts

```diff
@@ -152,33 +141,63 @@ export function ApolloServerPluginSchemaReporting(
           require('../../../package.json').version
         }`,
       };
+      let currentSchemaReporter: SchemaReporter | undefined;
 
-      logger.info(
-        'Apollo schema reporting starting! See your graph at ' +
-          `https://studio.apollographql.com/graph/${encodeURI(
-            graphRef,
-          )}/ with server info ${JSON.stringify(schemaReport)}`,
-      );
+      return {
+        schemaDidLoadOrUpdate({ apiSchema, coreSupergraphSdl }): void {
+          currentSchemaReporter?.stop();
 
-      const schemaReporter = new SchemaReporter({
-        schemaReport,
-        coreSchema,
-        apiKey: key,
-        endpointUrl,
-        logger,
-        // Jitter the startup between 0 and 10 seconds
-        initialReportingDelayInMs: Math.floor(
-          Math.random() * (initialDelayMaxMs ?? 10_000),
-        ),
-        fallbackReportingDelayInMs: 20_000,
-        fetcher,
-      });
+          if (overrideReportedSchema !== undefined) {
+            if (coreSupergraphSdl !== undefined) {
+              throw new Error(
+                [
+                  `The overrideReportedSchema option is incompatible with gateways`,
+                  `as the schema SDL is given directly by gateway. If you would`,
+                  `like to customize the schema SDL reported, please instead set`,
+                  `the option experimental_updateSupergraphSdl in your gateway`,
+                  `configuration.`,
+                ].join(' '),
+              );
+            }
+            logger.info(
+              'Apollo schema reporting: schema to report has been overridden',
```

Comment:
Will this print more than once ever?  No, because in the gateway case it's caught by the previous check?

Reply:
It wouldn't before, but I've now removed the check, so I've accordingly added code to ensure the hook early returns after executing once in the case when the schema is overridden.